packages/front-end/components/Experiment/ResultsDownloadButton.tsx

```diff
@@ -0,0 +1,121 @@
+import { ExperimentInterfaceStringDates } from "back-end/types/experiment";
+import { ExperimentReportResultDimension } from "back-end/types/report";
+import React, { useMemo } from "react";
+import { FaFileExport } from "react-icons/fa";
+import { useDefinitions } from "../../services/DefinitionsContext";
+import { ExperimentTableRow, getRisk } from "../../services/experiments";
+
+type UpdatedRow = {
+  metricName?: string;
+};
+
+export default function ResultsDownloadButton({
+  results,
+  experiment,
+}: {
+  results?: ExperimentReportResultDimension;
+  experiment: ExperimentInterfaceStringDates;
+}) {
+  const { getMetricById, ready } = useDefinitions();
+
+  const csvRows = [];
+
+  const rows = useMemo<ExperimentTableRow[]>(() => {
```

Comment:
I don't love that this bit of code is essentially running twice. Once here, and once inside of `CompactResults`. I attempted to pull it up to `Results` and just pass rows in to each component, but ran into issues.